device/common/include/traccc/seeding/device/impl/select_seeds.ipp

```diff
@@ -81,14 +81,14 @@ void select_seeds(
 
     device::doublet_counter_container_types::const_device
         doublet_counter_device(doublet_counter_container);
-    device_triplet_container triplet_device(triplet_view);
+    triplet_container_types::const_device triplet_device(triplet_view);
     seed_collection_types::device seed_device(seed_view);
 
     // Header of triplet: number of triplets per bin
     // Item of triplet: triplet objects per bin
     const unsigned int num_triplets_per_bin =
         triplet_device.get_headers().at(bin_idx).n_triplets;
-    const vecmem::device_vector<triplet> triplets_per_bin =
+    const vecmem::device_vector<const triplet> triplets_per_bin =
```

Comment:
Instead of the explicit type name, which is correct, we should use either

```c++
triplet_collection_types::const_device
```

, or better yet:

```c++
triplet_container_types::item_vector::const_reference
```

(I hope I wrote the correct types. I didn't try to compile this of course.) This way if the underlying types change, as long as the type definitions are correctly defined, this will keep working.

Reply:
`triplet_container_types::const_device::item_vector::const_reference`

I think this version is a bit overkill and goes against what we're using throughout all our device kernels. Isn't the point of having both the collection & container types declared to also be able to use them together coherently?

Reply:
(so I changed it to `triplet_collection_types::const_device`)

Reply:
Fine...